src/System.Windows.Forms.Design/src/System/ComponentModel/Design/CodeMarkers.cs

```diff
@@ -268,18 +260,17 @@ internal static byte[] StringToBytesZeroTerminated(string stringData)
             return data;
         }
 
-        public static byte[] AttachCorrelationId(byte[] buffer, Guid correlationId)
+        public static byte[]? AttachCorrelationId(byte[]? buffer, Guid correlationId)
         {
             if (correlationId == Guid.Empty)
             {
                 return buffer;
             }
 
-            byte[] correlationIdBytes = correlationId.ToByteArray();
-            byte[] bufferWithCorrelation = new byte[s_correlationMarkBytes.Length + correlationIdBytes.Length + (buffer is not null ? buffer.Length : 0)];
+            byte[] bufferWithCorrelation = new byte[s_correlationMarkBytes.Length + 16 + (buffer?.Length ?? 0)];
```

Comment:
Though it is 16 bytes length, I would keep `correlationIdBytes.Length` for readability. Or make it static for performance.

Reply:
Another thing we could imagine is to write
```suggestion
            byte[] bufferWithCorrelation = new byte[s_correlationMarkBytes.Length * 2 + (buffer?.Length ?? 0)];
```
as `s_correlationMarkBytes` also contains Guid bytes. Would that be enough?